Sources/KovaleeSurvey/KovaleeSurveyManagerImpl.swift

```diff
@@ -0,0 +1,51 @@
+import Survicate
+import KovaleeSDK
+import Foundation
+import KovaleeFramework
+
+class KovaleeSurveyManagerImpl: SurveyManager, Manager {
+    
+    private let survicate: SurvicateSdk?
+
+    init(withKey workspaceKey: String?) {
+
+        if let workspaceKey {
```

Comment:
A couple of implementation details.

1. always prefer guard instead of if in cases like this one
```
guard let workspaceKey else {
     KLogger.error("No workspaceKey found for Survicate. Survicate was not configured.")
     return
}
```

2. we need to check for any exception here
```
do {
   try self.survicate?.setWorkspaceKey(workspaceKey)
   self.survicate?.initialize()
   self.survicate?.addListener(self)
} catch {
    KLogger.error("Couldn't initialise Survicate: \(error)")
}
```

Reply:
🫡